app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,52 @@
+class PostsController < ApplicationController
+
+  before_action :authenticate_user!, :except => [:index, :show]
+
+  def new
+    @post = Post.new
+  end
+
+  def index
+    if user_signed_in?
```

Comment:
Style/IfUnlessModifier: Favor modifier `if` usage when having a single-line body. Another good alternative is the usage of control flow `&&`/`||`. (https://github.com/bbatsov/ruby-style-guide#if-as-a-modifier)